src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java

```diff
@@ -203,8 +210,47 @@ private static boolean shouldSchedulePatronNotices(LostItemFeeRefundContext cont
     final Result<CqlQuery> fetchQuery = exactMatch("loanId", context.getLoan().getId())
       .combine(exactMatchAny("feeFineType", feeFineTypes), CqlQuery::and);
 
-    return accountRepository.findAccountsAndActionsForLoanByQuery(fetchQuery)
-      .thenApply(r -> r.map(context::withAccounts));
+      return accountRepository.findAccountsAndActionsForLoanByQuery(fetchQuery)
+        .thenApply(r -> r.map(this::filterAccountsForRefund))
+        .thenApply(r -> r.map(context::withAccounts));
+  }
+
+  private Collection<Account> filterAccountsForRefund(Collection<Account> accounts) {
+    Account latestLostItemFeeAccount = getLatestAccount(accounts, LOST_ITEM_FEE_TYPE);
+
+    if (latestLostItemFeeAccount != null
```

Comment:
Have you considered org.apache.commons.lang3.ObjectUtils.allNotNull() ?

Reply:
@viktor-draban using `allNotNull()` will cause NPE if `latestLostItemFeeAccount` is null

Reply:
@roman-barannyk please consider extracting such complex conditions into separate methods